diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2dc33ef4..cf3d7b203 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,12 +17,12 @@ if (NOT DOLFIN_VERSION_RELEASE)
 endif()
 
 #------------------------------------------------------------------------------
-# Require and use C++11
+# Require and use C++14
 
-# Use C++11
-set(CMAKE_CXX_STANDARD 11)
+# Use C++14
+set(CMAKE_CXX_STANDARD 14)
 
-# Require C++11
+# Require C++14
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Do not enable compler-specific extensions
@@ -31,11 +31,11 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 #------------------------------------------------------------------------------
 # Check compiler version
 
-# Check for GCC version - earlier versions have insuffcient C++11
+# Check for GCC version - earlier versions have insuffcient C++14
 # support, or bugs.
 if (CMAKE_COMPILER_IS_GNUCXX)
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8.3)
-    message(FATAL_ERROR "GCC version must be at least 4.8.3 (for sufficient C++11  support and to avoid some compiler bugs). You have version ${CMAKE_CXX_COMPILER_VERSION}")
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.1.0)
+    message(FATAL_ERROR "GCC version must be at least 6.1.0 (for sufficient C++14  support and to avoid some compiler bugs). You have version ${CMAKE_CXX_COMPILER_VERSION}")
   endif()
 endif()
 
