diff --git a/dolfin/la/EigenLUSolver.h b/dolfin/la/EigenLUSolver.h
index 12d9fc365..876baeda4 100644
--- a/dolfin/la/EigenLUSolver.h
+++ b/dolfin/la/EigenLUSolver.h
@@ -83,9 +83,10 @@ namespace dolfin
     std::string parameter_type() const
     { return "lu_solver"; }
 
-  private:
     // Eigen LU implementation class
     class EigenLUImplBase;
+
+  private:
     std::unique_ptr<EigenLUImplBase> _impl;
 
     // Available LU solvers and descriptions
